core/src/test/scala/pink/cozydev/lucille/DefaultBooleanAndSuite.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 CozyDev
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pink.cozydev.lucille
+
+import pink.cozydev.lucille.Query._
+
+class DefaultBooleanAndSuite extends munit.FunSuite {
+
+  val parser = new QueryParser(defaultBooleanOR = false)
```

Comment:
I do find this `defaultBooleanOR = false` a bit awkward to work with implying that we have a parser with defaultBooleanAnd-like behaviour

Reply:
I added a convenience method to the companion object so this now looks like:
```scala
val parser = QueryParser.withDefaultOperatorAND
```

Which is a bit more explicit.